packages/discovery-provider/src/api/v1/tracks.py

```diff
@@ -455,6 +455,161 @@ def get(self, track_id):
         abort_not_found(track_id, ns)
 
 
+stream_url_parser = reqparse.RequestParser(argument_class=DescriptiveArgument)
+stream_url_parser.add_argument(
+    "preview",
+    description="""Optional - true if streaming track preview""",
+    type=inputs.boolean,
+    required=False,
+    default=False,
+)
+stream_url_parser.add_argument(
+    "user_signature",
+    description="""Optional - signature from the requesting user's wallet.
+        This is needed to authenticate the user and verify access in case the track is gated.""",
+    type=str,
+)
+stream_url_parser.add_argument(
+    "user_data",
+    description="""Optional - data which was used to generate the optional signature argument.""",
+    type=str,
+)
+stream_url_parser.add_argument(
+    "nft_access_signature",
+    description="""Optional - gated content signature for this track which was previously generated by a registered DN.
+        We perform checks on it and pass it through to CN.""",
+    type=str,
+)
+stream_url_parser.add_argument(
+    "skip_play_count",
+    description="""Optional - boolean that disables tracking of play counts.""",
+    type=bool,
+    required=False,
+    default=False,
+)
+stream_url_parser.add_argument(
+    "api_key",
+    description="""Optional - API key for third party apps. This is required for tracks that only allow specific API keys.""",
+    type=str,
+    required=False,
+    default=None,
+)
+
+
+@ns.route("/<string:track_id>/stream-url")
+class TrackStreamUrl(Resource):
```

Comment:
What you think about adding a query param to existing stream endpoint: `json=true` that will return JSON instead of 302.

This would save us from copy-pasting existing endpoint.

Reply:
Yep, in hindsight the query param idea was better but like Ray said, it doesn't really matter for this change. 
For now this is just to test out our theory. If its something we want to go with long-term I'm thinking we would redesign to not use this url at all and we can just delete it (I imagine we would put it somewhere we already get track info)